test/csuite/parse_opts/main.c

```diff
@@ -0,0 +1,315 @@
+        while ((ch = __wt_getopt(opts->progname, argc, argv, "b:c:def:P:T:v")) != EOF)
+            switch (ch) {
+            case 'c':
+                x_opts->c_option = __wt_optarg;
+                break;
+            case 'd':
+                x_opts->d_option = true;
+                break;
+            case 'e':
+                x_opts->e_option = true;
+                break;
+            case 'f':
+                x_opts->f_option = atoi(__wt_optarg);
+                break;
+            default:
+                if (testutil_parse_opt(opts, ch) != 0) {
+                    (void)fprintf(stderr, "usage: %s%s%s\n", opts->progname, x_usage, opts->usage);
+                    testutil_assert(false);
+                }
+            }
+        testutil_parse_opt_end(opts);
+    }
+}
+
+/*
+ * verify_expect --
+ *     Verify the returned options against the expected options.
+ */
+static void
+verify_expect(TEST_OPTS *opts, EXTENDED_OPTS *x_opts, TEST_OPTS *expect, EXTENDED_OPTS *x_expect)
+{
+#define VERIFY_INT(o, e, field)         \
+    if (o->field != 0 || e->field != 0) \
+    testutil_assert(o->field == e->field)
+#define VERIFY_STR(o, e, field)                               \
+    do {                                                      \
+        if (o->field != NULL || e->field != NULL) {           \
+            testutil_assert(o->field != NULL);                \
+            testutil_assert(e->field != NULL);                \
+            testutil_assert(strcmp(o->field, e->field) == 0); \
+        }                                                     \
+    } while (0)
+
+    /*
+     * opts->home is always set, even without -h on the command line, so don't check it here.
+     */
+    VERIFY_STR(opts, expect, build_dir);
+    VERIFY_STR(opts, expect, tiered_storage_source);
+    VERIFY_INT(opts, expect, table_type);
+    VERIFY_INT(opts, expect, do_data_ops);
+    VERIFY_INT(opts, expect, preserve);
+    VERIFY_INT(opts, expect, tiered_storage);
+    VERIFY_INT(opts, expect, verbose);
+    VERIFY_INT(opts, expect, nrecords);
+    VERIFY_INT(opts, expect, nops);
+    VERIFY_INT(opts, expect, nthreads);
+    VERIFY_INT(opts, expect, n_append_threads);
+    VERIFY_INT(opts, expect, n_read_threads);
+    VERIFY_INT(opts, expect, n_write_threads);
+
+    VERIFY_STR(x_opts, x_expect, c_option);
+    VERIFY_INT(x_opts, x_expect, d_option);
+    VERIFY_INT(x_opts, x_expect, e_option);
+    VERIFY_INT(x_opts, x_expect, f_option);
+}
+
+/*
+ * cleanup --
+ *     Clean up allocated resources.
+ */
+static void
+cleanup(TEST_OPTS *opts, EXTENDED_OPTS *x_opts)
+{
+    (void)x_opts; /* Nothing to clean up here. */
+
+    testutil_cleanup(opts);
+}
+
+/*
+ * main --
+ *     Unit test for test utility functions.
+ */
+int
+main(int argc, char *argv[])
+{
+    TEST_OPTS *expect, opts;
+    EXTENDED_OPTS *x_expect, x_opts;
+    size_t i;
+    int nargs;
+    char *const *cmd;
+
+    if (argc > 1) {
```

Comment:
The `check` function does not support anything other than `parse_opt(s)` so should that bad usage get caught here instead? I.e. running it from the command line cannot take any other argument.

Reply:
Fixed